module/data/item/templates/activated-effect.mjs

```diff
@@ -0,0 +1,106 @@
+        units: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.TargetUnits"}),
+        type: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.TargetType"})
+      }, {label: "DND5E.Target"}),
+      range: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.RangeNormal"}),
+        long: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.RangeLong"}),
+        units: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.RangeUnits"})
+      }, {label: "DND5E.Range"}),
+      uses: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.LimitedUsesAvailable"}),
+        max: new FormulaField({required: true, deterministic: true, label: "DND5E.LimitedUsesMax"}),
+        per: new foundry.data.fields.StringField({
+          required: true, blank: false, nullable: true, initial: null, label: "DND5E.LimitedUsesPer"
+        }),
+        recovery: new FormulaField({required: true, label: "DND5E.RecoveryFormula"})
+      }, {label: "DND5E.LimitedUses"}),
+      consume: new foundry.data.fields.SchemaField({
+        type: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.ConsumeType"}),
+        target: new foundry.data.fields.StringField({
+          required: true, blank: false, nullable: true, initial: null, label: "DND5E.ConsumeTarget"
+        }),
+        amount: new foundry.data.fields.NumberField({required: true, label: "DND5E.ConsumeAmount"})
+      }, {label: "DND5E.ConsumeTitle"})
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static migrateData(source) {
+    this.migrateMaxUses(source);
+    this.migrateRanges(source);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Ensure a 0 in max uses is converted to an empty string rather than "0".
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   */
+  static migrateMaxUses(source) {
+    if ( (source.uses?.max === 0) || (source.uses?.max === "0") ) source.uses.max = "";
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fix issue with some imported range data that uses the format "100/400" in the range field,
+   * rather than splitting it between "range.value" & "range.long".
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   */
+  static migrateRanges(source) {
+    if ( typeof source.range?.value !== "string" ) return;
+    const split = source.range.value.split("/");
+    if ( !Number.isNaN(split[0]) ) source.range.value = Number(split[0]);
+    if ( (split.length > 1) && !Number.isNaN(split[1]) ) source.range.long = Number(split[1]);
```

Comment:
We should use `Number.isNumeric`. `Number.isNaN` only checks for the `NaN` value explicitly, which these values can't be because they're strings.

```suggestion
    const [value, long] = source.range.value.split("/");
    if ( Number.isNumeric(value) ) source.range.value = Number(value);
    if ( Number.isNumeric(long) ) source.range.long = Number(long);
```